### PR TITLE
expression: push down json_valid to tikv

### LINKED ISSUE #38289: Support json_valid push down to TiKV
## Enhancement
Now TiDB support some json function to TiKV : json_extract
We should also support json_valid function to tikv

### PR DESCRIPTION
**Pull request by lizhenhuan**
Signed-off-by: lizhenhuan <1916038084@qq.com>

### What problem does this PR solve?

Issue Number:  close #38289

Problem Summary:
Implement json_valid function push down to tikv

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Push dow json_valid to tikv
```

### CODE REVIEW

**Comment by bb7133** [comment]
Thanks for the contribution! @xiongjiwei @YangKeao Do you think it's a good time to push this func to TiKV?

File: planner/core/integration_test.go [hunk 1]
```diff
@@ -3064,7 +3064,7 @@ func TestScalarFunctionPushDown(t *testing.T) {
 	tk := testkit.NewTestKit(t, store)
 	tk.MustExec("use test")
 	tk.MustExec("create table t(id int signed, id2 int unsigned, c varchar(11), d datetime, b double, e bit(10))")
-	tk.MustExec("insert into t(id, id2, c, d) values (-1, 1, 'abc', '2021-12-12')")
+	tk.MustExec("insert into t(id, id2, c, d) values (-1, 1, '{\"a\":1}', '2021-12-12')")
```
Comment (TonsnakeLin) [inline planner/core/integration_test.go:3067]: Could you add more cases for json_valid?

**Comment by YangKeao** [comment]
As a reference for other reviewer, the corresponding PR on TiKV: https://github.com/tikv/tikv/pull/13572 has been merged.

**Review by YangKeao — APPROVED** [review]
LGTM

File: planner/core/integration_test.go [hunk 2]
```diff
@@ -3375,6 +3375,10 @@ func TestScalarFunctionPushDown(t *testing.T) {
 	tk.MustQuery("explain analyze select /*+read_from_storage(tikv[t])*/ * from t where ascii(e);").
 		CheckAt([]int{0, 3, 6}, rows)
 
+	rows[1][2] = "eq(json_valid(test.t.c), 1)"
+	tk.MustQuery("explain analyze select /*+read_from_storage(tikv[t])*/ * from t where json_valid(c)=1;").
+		CheckAt([]int{0, 3, 6}, rows)
+
```
Comment (YangKeao) [inline planner/core/integration_test.go:3381]: It seems that this is not an empty line, but contains tab/space. Please delete them (or simply format the codes).

Comment (lizhenhuan) [reply planner/core/integration_test.go]: I change this empty line

**Review by sticnarf — APPROVED** [review]

File: expression/expression.go [hunk 3]
```diff
@@ -1072,7 +1072,7 @@ func scalarExprSupportedByTiKV(sf *ScalarFunction) bool {
 		ast.JSONType, ast.JSONExtract, ast.JSONObject, ast.JSONArray, ast.JSONMerge, ast.JSONSet,
 		ast.JSONInsert /*ast.JSONReplace,*/, ast.JSONRemove, ast.JSONLength,
 		// FIXME: JSONUnquote is incompatible with Coprocessor
```
Comment (xhebox) [inline expression/expression.go:1074]: LGTM, I think we can remove L1074

Comment (lizhenhuan) [reply expression/expression.go]: Removed

**Review by xhebox — APPROVED** [review]

**Pull request merged**